9.1

LetaTreiden opened the pull request.

LetaTreiden committed:
9.1

LetaTreiden committed:
9.2

корректные коды

LetaTreiden committed:
9.3

обработка ошибок, статусы

LetaTreiden committed:
9.4

LetaTreiden committed:
9.5 

проверка тестов, переписан метод в юзер сервис

LetaTreiden committed:
9.6

LetaTreiden committed:
9.7

правки в путях

LetaTreiden committed:
9.7.1

исправлены названия перемннных

LetaTreiden committed:
9.9

переименования и правки в ошибках и статусах

LetaTreiden committed:
9.10

для прохождения тестов

LetaTreiden committed:
Delete Error.java

LetaTreiden committed:
9.10

для тествов

LetaTreiden committed:
9.11

пом

LetaTreiden committed:
Add files via upload

LetaTreiden committed:
9.12

LetaTreiden committed:
Merge branch 'main' into add-friends-likes

LetaTreiden committed:
Delete target directory

LetaTreiden committed:
Delete Error.java

LetaTreiden committed:
Delete exceptionHandler.java

LetaTreiden committed:
9.13

обработка ошибки поиска

LetaTreiden committed:
9.14

обавлены методы получения фильма и юзера

LetaTreiden committed:
9.15

jib,rb b bcrk.xtybz

LetaTreiden committed:
9.16

ошибка в гет рэйт

LetaTreiden committed:
9.16.1

ошибка в гет рэйт 2

LetaTreiden committed:
9.17

правка в добавлении друзей

LetaTreiden committed:
9.18

метод добавления в друзья возвратный

LetaTreiden committed:
9.19

LetaTreiden committed:
9.20

корректное отображение друзей и пользователей

LetaTreiden committed:
9.21

класс валидации, перенос действий в классы сервиса

LetaTreiden committed:
9.21

изменение логики валидации

LetaTreiden committed:
9.23

LetaTreiden committed:
9.24

LetaTreiden committed:
9.25

валидация и обработка ошибок

LetaTreiden committed:
9.26

get rates

LetaTreiden committed:
9.27

не существующий пользователь

LetaTreiden committed:
9.28

несущесвутющий пользователь

LetaTreiden committed:
9.29

Неизвестный друг

LetaTreiden committed:
9.30

добавление и удаление друзей

LetaTreiden committed:
9.???

гет фильм для тестов

LetaTreiden committed:
обработка исключений в получении пользователя поid

LetaTreiden committed:
что-то лишнее

LetaTreiden committed:
что-то не лишнее

LetaTreiden committed:
вывод общих друзей

LetaTreiden committed:
удаление друзей

LetaTreiden committed:
удаление друзей2

LetaTreiden committed:
Способ хранения пользователей

LetaTreiden committed:
откат

LetaTreiden committed:
Вывод друзей

LetaTreiden committed:
вывод списка друзей

должен появляться новые полный список

LetaTreiden committed:
Нет общих друзей

LetaTreiden committed:
mutual

LetaTreiden committed:
неи общих друзей

LetaTreiden committed:
Add files via upload

LetaTreiden committed:
нет общих друзей ошибка

LetaTreiden committed:
нет ошибки нет общих друзей

LetaTreiden committed:
Set на List в общих друзьях

LetaTreiden committed:
удаление из друзей

LetaTreiden committed:
последняя правка

редакция строчек, удаление лишнего кода
чтобы все красиво и читаемо

kukolev commented on src/main/java/ru/yandex/practicum/filmorate/validation/Validate.java:
```diff
@@ -0,0 +1,85 @@
+import java.time.LocalDate;
+
+@Component
+public class Validate {
```
Некорректное название класса. Validate это глагол, а классы принято назвать как существительные.

kukolev commented on src/main/java/ru/yandex/practicum/filmorate/validation/Validate.java:
```diff
@@ -0,0 +1,85 @@
+public class Validate {
+    private static final Logger logger = LoggerFactory.getLogger(Film.class);
+    private static final LocalDate FIST_RELEASE_DATE = LocalDate.of(1895, 12, 28);
+    String string;
```
Нарушен принцип инкапсуляции. Поля класса не должны быть видимы другим классам. Лучше сделать поле приватным и сделать геттер и сеттер для полей, если нужен доступ извне.

kukolev commented on src/main/java/ru/yandex/practicum/filmorate/validation/Validate.java:
```diff
@@ -0,0 +1,85 @@
+    }
+
+    public boolean isFilmExist(Film film, InMemoryFilmStorage filmStorage) throws NotFoundException {
+        System.out.println(film);
```
Вместо вывода в консоль нужно использовать логирование, как это сделано в остальных классах.

LetaTreiden committed:
Правки с ревью

класс валидации - переменная стринг приватная, 
- изменено название класса
- замена саут на логгирование

kukolev commented on src/main/java/ru/yandex/practicum/filmorate/validation/Validator.java:
```diff
@@ -0,0 +1,85 @@
+public class Validator {
+    private static final Logger logger = LoggerFactory.getLogger(Film.class);
+    private static final LocalDate FIST_RELEASE_DATE = LocalDate.of(1895, 12, 28);
+    private static String string;
```
Поле класса для этой строки делать нет смысла, т.к. как поле оно нигде не используется. Можно заменить поле на переменную в каждом методе и все будет также работать. 

Также при работе нескольких пользователей одновременно здесь возможны проблемы, т.к. поле статическое - строка, записанная в методе validateFilm первого пользователя может быть перезаписана в методе isUserExist второго пользователя.

LetaTreiden committed:
правки ревью
